client/lib/track-element-size/README.md

```diff
@@ -0,0 +1,90 @@
+Track Element Size
+==================
+
+This module is designed to provide utilities for tracking an element's size in the DOM.
+
+## Usage
+
+### `useWindowResizeRect`
+
+`useWindowResizeRect` is a React hook that tracks the size of an element's bounding client rect as
+the window is resized. It doesn't track modifications due to other reasons, such as layout changes.
+
+It returns both a ref, to be applied on the element to be measured, and a rect with the element's
+dimensions. This rect will be null at first, but will be updated with the initial values on a later
+render, as well as kept up to date after any window resizes.
+
+Here's a small sample:
+
+```JSX
+import { useWindowResizeRect } from 'lib/track-element-size';
+
+function SampleComponent() {
+  // Get ref, rect and subscribe to changes.
+  const [ ref, rect ] = useWindowResizeRect();
```

Comment:
I don't see why not. Each hook instance gets its own state and callback. It might be inefficient, due to the multiple state updates, but it should be safe.

Is there anything in particular you think could fail here?

Reply:
Nothing in particular, just curious 🙂 But even if there is one resize event and one layout reflow, this will lead to two state updates, right? Nothing wrong with that, just checking my understanding.

Reply:
Yes, that's correct 🙂 

If you think that having multiple rects being watched in the same component could be a common use-case, I'm happy to try to find some way of optimising it. Although to be fair, I'm not even sure where I'd start... Hooks are designed to be used multiple times, completely independently, so I suspect it would be hard to break that abstraction and batch updates somehow.